new-lamassu-admin/src/pages/Blacklist/BlacklistAdvanced.js

```diff
@@ -0,0 +1,177 @@
+import { makeStyles } from '@material-ui/core/styles'
+import { Form, Formik, Field } from 'formik'
+import * as R from 'ramda'
+import React, { useState } from 'react'
+import * as Yup from 'yup'
+
+import ErrorMessage from 'src/components/ErrorMessage'
+import Modal from 'src/components/Modal'
+import { ActionButton, IconButton, Button } from 'src/components/buttons'
+import { TextInput } from 'src/components/inputs/formik'
+import DataTable from 'src/components/tables/DataTable'
+import { ReactComponent as DisabledDeleteIcon } from 'src/styling/icons/action/delete/disabled.svg'
+import { ReactComponent as DeleteIcon } from 'src/styling/icons/action/delete/enabled.svg'
+import { ReactComponent as EditIcon } from 'src/styling/icons/action/edit/enabled.svg'
+import { ReactComponent as DefaultIconReverse } from 'src/styling/icons/button/retry/white.svg'
+import { ReactComponent as DefaultIcon } from 'src/styling/icons/button/retry/zodiac.svg'
+
+import styles from './Blacklist.styles'
+
+const useStyles = makeStyles(styles)
+
+const DEFAULT_MESSAGE = `This address may be associated with a deceptive offer or a prohibited group. Please make sure you're using an address from your own wallet.`
+
+const getErrorMsg = (formikErrors, formikTouched, mutationError) => {
+  if (!formikErrors || !formikTouched) return null
+  if (mutationError) return 'Internal server error'
+  if (formikErrors.event && formikTouched.event) return formikErrors.event
+  if (formikErrors.message && formikTouched.message) return formikErrors.message
+  return null
+}
+
+const BlacklistAdvanced = ({
+  data,
+  editBlacklistMessage,
+  onClose,
+  mutationError
+}) => {
+  const classes = useStyles()
+  const [selectedMessage, setSelectedMessage] = useState(null)
+
+  const elements = [
+    {
+      name: 'label',
+      header: 'Label',
+      width: 250,
+      textAlign: 'left',
+      size: 'sm',
+      view: it => R.path(['label'], it)
+    },
+    {
+      name: 'content',
+      header: 'Content',
+      width: 690,
+      textAlign: 'left',
+      size: 'sm',
+      view: it => R.path(['content'], it)
+    },
+    {
+      name: 'edit',
+      header: 'Edit',
+      width: 130,
+      textAlign: 'center',
+      size: 'sm',
+      view: it => (
+        <IconButton
+          className={classes.deleteButton}
+          onClick={() => setSelectedMessage(it)}>
+          <EditIcon />
+        </IconButton>
+      )
+    },
+    {
+      name: 'deleteButton',
+      header: 'Delete',
+      width: 130,
+      textAlign: 'center',
+      size: 'sm',
+      view: it => (
+        <IconButton
+          className={classes.deleteButton}
+          disabled={
+            !R.isNil(R.path(['allowToggle']) && !R.path(['allowToggle'], it))
```

Comment:
Hmmm this line is dubious...

`it` missing in the first `R.path()`, and it seems like it should be `!isNil(allowToggle) && !allowToggle`, right?